### PR TITLE
feat: ushiftRight bitblasting

### PR DESCRIPTION
**Pull request by bollu**
This adds theorems `ushiftRight_rec_zero`, `ushiftRight_rec_succ`, `ushiftRight_rec_eq`, and `shiftRight_eq_shiftRight_rec`.

### CODE REVIEW

File: src/Init/Data/BitVec/Lemmas.lean [hunk 1]
```diff
@@ -731,6 +731,17 @@ theorem getLsb_shiftLeft' {x : BitVec w₁} {y : BitVec w₂} {i : Nat} :
     getLsb (x >>> i) j = getLsb x (i+j) := by
   unfold getLsb ; simp
 
+@[simp]
+theorem ushiftRight_zero_eq (x : BitVec w) : x >>> 0 = x := by
+
+  simp [bv_toNat]
```
Comment (tobiasgrosser) [inline src/Init/Data/BitVec/Lemmas.lean:737]:
```suggestion
  simp [bv_toNat]
```

**Comment by tobiasgrosser** [comment]
Nice. Let's wait for @alexkeizer feedback and then this can be undrafted.

File: src/Init/Data/BitVec/Bitblast.lean [hunk 2]
```diff
@@ -431,4 +427,67 @@ theorem shiftLeft_eq_shiftLeftRec (x : BitVec w₁) (y : BitVec w₂) :
...
+and thus `x >>> (y ||| z) = x >>> (y + z) = x >>> y >>> z`.
+-/
+theorem ushiftRight'_or_of_and_eq_zero {x : BitVec w₁} {y z : BitVec w₂}
+    (h : y &&& z = 0#w₂) :
+    x >>> (y ||| z) = x >>> y >>> z := by
+  simp [← add_eq_or_of_and_eq_zero _ _ h, toNat_add_of_and_eq_zero h, shiftRight_add]
+
+theorem ushiftRightRec_eq (x : BitVec w₁) (y : BitVec w₂) (n : Nat) :
+  ushiftRightRec x y n = x >>> (y.truncate (n + 1)).zeroExtend w₂ := by
+  induction n generalizing x y
```
Comment (tobiasgrosser) [inline src/Init/Data/BitVec/Bitblast.lean:469]:
```suggestion
theorem ushiftRightRec_eq (x : BitVec w₁) (y : BitVec w₂) (n : Nat) :
    ushiftRightRec x y n = x >>> (y.truncate (n + 1)).zeroExtend w₂ := by
  induction n generalizing x y
```

File: src/Init/Data/BitVec/Lemmas.lean [hunk 3]
```diff
@@ -731,6 +731,17 @@ theorem getLsb_shiftLeft' {x : BitVec w₁} {y : BitVec w₂} {i : Nat} :
...
+
+/-! ### ushiftRight reductions from BitVec to Nat -/
+
+@[simp]
+theorem ushiftRight_eq' (x : BitVec w₁) (y : BitVec w₂) :
+  x >>> y = x >>> y.toNat := by rfl
+
```
Comment (tobiasgrosser) [inline src/Init/Data/BitVec/Lemmas.lean:744]:
```suggestion
    x >>> y = x >>> y.toNat := by rfl

```

**Comment by semorrison** [comment]
LGTM.

**Comment by bollu** [comment]
@semorrison I'm not going to mark it `awaiting-review`, since I received an LGTM. I guess you'll merge it in later :)

**Comment by hargoniX** [comment]
Review from Alex no longer necessary after some internal communication.

**Pull request merged**